File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

char* solution(const char* cards1[], size_t cards1_len, const char* cards2[], size_t cards2_len, const char* goal[], size_t goal_len) {
    char* answer = (char*)malloc(sizeof(char)*4);
    int i = 0;
    int j = 0;
    int k = 0;
    bool possible = true;
    while(k < goal_len){
        if(i<cards1_len && strcmp(goal[k], cards1[i]) == 0){i++;k++;}
        else if(j<cards2_len && strcmp(goal[k],cards2[j]) == 0){j++;k++;}
        else{possible = false;break;}
    }

    if(possible){strcpy(answer, "Yes");}
    else{strcpy(answer, "No");}

    return answer;
}

File: 프로그래머스/1/159994.카드 뭉치-Python/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = ''
    i1 = 0
    i2 = 0
    answer = "Yes"
    for g in goal:

        if i1 < len(cards1) and cards1[i1] == g:
            i1 += 1
        elif i2 < len(cards2) and cards2[i2] == g:
            i2 += 1
        elif i1 >= len(cards1) and i2 >= len(cards2):
            answer = "Yes"
        else:
            answer = "No"
            break


    return answer